### ISSUE TITLE
[Enhancement] Would it be nice for AppiumLocalService to download appium server 

### ISSUE THREAD

**Issue by GiannisPapadakis** [appium/java-client#248]
We could have implementation in AppiumLocalService to download appium server before searching for  Appium installation from:https://bitbucket.org/appium/appium.app/downloads/. Would it be nice to have automatic download of the server before starting via node the server instance?

**Comment by TikhomirovSergey** [comment]
It is the interesting idea :)

I think that the workflow should be like that:
__1)__ AppiumLocalService checks the given node server. If it has been defined properly or there is an instance which has been set up via npm then it starts the server.  This step is implemented and you use it with the current release.

__2)__ If there is no appium node server it tries install it via npm. If this step is successful it goes to __1)__, else it throws the exception.

@GiannisPapadakis What do you think about this.
@Jonahss I think that this idea is interesting for you too :)

**Comment by GiannisPapadakis** [comment]
I think yes it would be a good feature to install Appium automatically, if not installed, via npm before starting the server.

**Comment by Jonahss** [comment]
But appium isn't easily installed just by downloading :(
There are environment variables needed for Android and iOS. And it relies
on Android SDKs being installed and other things like that.

On Thu, Sep 17, 2015 at 1:35 AM, Giannis Papadakis <notifications@github.com
